lib/test.go

```diff
@@ -291,3 +291,18 @@ func ReceiveBallot(t *testing.T, nodeRunner *NodeRunner, ballot *Ballot) error {
 	err = nodeRunner.handleBallotMessage(ballotMessage)
 	return err
 }
+
+type TestBroadcastor struct {
+	Messages map[string]common.Message
+}
+
+func NewTestBroadcastor() TestBroadcastor {
+	p := TestBroadcastor{}
+	p.Messages = make(map[string]common.Message)
+	return p
+}
+
+func (b TestBroadcastor) Broadcast(_ *network.ConnectionManager, message common.Message) (errs map[string]error) {
```

Comment:
It does not matter but the Broadcaster needs ConnectManager when it broadcast? :-) 

```
type Broadcastor interface {
	Broadcast(common.Message) (errs map[string]error)
}
```

Reply:
@anarcher Is there any problem?? :)

Reply:
@Charleslee522  No, Not really :-) Just MHO :-)

Reply:
@anarcher Oh, I just wanted to know your humble opinion :)

Reply:
In broadcast function, c.connected(address) and c.GetConnection are needed :)
```
func (b SimpleBroadcastor) Broadcast(c *ConnectionManager, message common.Message) (errs map[string]error) {
	for addr, _ := range c.connected {
		go func(v *node.Validator) {
			if v == nil {
				panic("Validator connected but not registered")
			}

			client := c.GetConnection(v.Address())

			var err error
			if message.GetType() == BallotMessage {
				_, err = client.SendBallot(message)
			} else if message.GetType() == string(TransactionMessage) {
				_, err = client.SendMessage(message)
			} else {
				panic("invalid message")
			}

			if err != nil {
				errs[v.Address()] = err
			}
		}(c.validators[addr])
	}
	return
}
```

Reply:
It meant like below: 
```
type Broadcastor interface {
	Broadcast(common.Message) (errs map[string]error)
}

type SimpleBroadcastor struct {
	cm *ConnectionManager
}
func NewSimpleBroadcastor(cm *ConnectionManager) *SimpleBroadcastor

type MultiConnManagerBroadcastor struct { ... }

type GossipBroadcastor struct { ... } 

type StaticBroadcastor 
type TestBroadcastor 

```
But If we need to  choose `ConnectionManager` in  a request scope, We should use  arguments like  `(c *ConnectionManager,msg common.Message) `

Reply:
@anarcher Thank you for giving me a good idea 👍